### PR TITLE
Implement events passing to imgui shell

### PR DESCRIPTION
**Pull request by dragonnn**
This commits adds:
- passing events from process_input_event into imgui shell when they are not captured by any window
- simple button with callback shell item

Imgui was complete moved from OutputSurfaceData in backends directly into Output to archive it.
A new event Shell was add too ConfigEvent for passing callbacks from imgui to ConfigVM when a script action is taken (for example from a button).
Needs this to work: https://github.com/Smithay/smithay/pull/457

### CODE REVIEW

File: src/framework/backend/udev.rs [hunk 1]
```diff
@@ -267,9 +267,8 @@ struct OutputSurfaceData {
     surface: RenderSurface,
     _render_timer: RenderTimerHandle,
     fps: fps_ticker::Fps,
-    imgui: Option<imgui::SuspendedContext>,
-    imgui_pipeline: imgui_smithay_renderer::Renderer,
-
+    //imgui: Option<imgui::SuspendedContext>,
+    //imgui_pipeline: imgui_smithay_renderer::Renderer,
```
Comment (PolyMeilex) [inline src/framework/backend/udev.rs]: Let's just remove this, if it is no longer used

**Comment by PolyMeilex** [comment]
Ok, looks good to me.
But it needs a small rebase, after Smithay update, transformations got flipped, so now X11 and UDev have `Transform::Normal`, and winit is `Transform::Flipped180` it looks like this  change causes some conflicts in your code.

File: src/input_handler.rs [hunk 2]
```diff
@@ -23,15 +23,26 @@ impl Anodium {
...
     ) {
-        match event {
+        let captured = match &event {
             InputEvent::Keyboard { event, .. } => {
                 let action = self.keyboard_key_to_action::<I>(event);
-                self.shortcut_handler(action)
+                if action == KeyAction::Filtred {
+                    true
+                } else {
+                    if action != KeyAction::None {
```
Comment (PolyMeilex) [inline src/input_handler.rs]: this can be collapsed to else if

File: src/framework/backend/udev.rs (hunk 1 above)
Comment (dragonnn) [reply src/framework/backend/udev.rs]: sure, forgot about this, fixed

File: src/input_handler.rs (hunk 2 above)
Comment (dragonnn) [reply src/input_handler.rs]: fixed

**Comment by dragonnn** [comment]
Will do the rebase soon :).

**Comment by dragonnn** [comment]
Ok, that should now be fine.

**Pull request merged**